public/javascripts/SVValidate/src/mission/Mission.js

```diff
@@ -103,6 +105,33 @@ function Mission(params) {
         svv.statusField.setProgressText(completionRate);
     }
 
+    /**
+     * Updates status bar (UI) and current mission properties whenever undo is pressed.
+     */
+    function updateMissionProgressUndo() {
```

Comment:
Similar to how I've had you do in other places, can you combine this with the `updateMissionProgress()` function and just add a parameter to indicate whether you are updating because of an undo?